hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/rpc/RpcClient.java

```diff
@@ -1440,6 +1443,7 @@ public void close() throws IOException {
     IOUtils.cleanupWithLogger(LOG, ozoneManagerClient, xceiverClientManager);
     keyProviderCache.invalidateAll();
     keyProviderCache.cleanUp();
+    clientMetrics.unregister();
```

Comment:
If there are multiple clients in the same JVM, the first one to `unregister()` will break all other clients' metrics.  I think the singleton instance should be reference-counted, and unregistered only for the last one.  (Alternatively, maybe it should not be a singleton.)